### PR TITLE
Sea zone string checks

### PR DESCRIPTION
**Pull request by frigoref**
-Introducing method isTerritoryNameIndicatingWater
-Common findTerritoryName code moved to new method in ui.Util
-References to "Sea Zone" changed to refer to new ui.Util constant

Maybe we need a new class for a "territory name -> polygons"-map (Map<String, List<Polygon>> terrPolygons). What are your throughts on that?

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/659)

### CODE REVIEW

File: src/games/strategy/ui/Util.java [hunk 1]
```diff
@@ -24,6 +30,8 @@
 import games.strategy.util.EventThreadJOptionPane;
 
 public final class Util {
+  public final static String territorySeaZoneInfix = "Sea Zone";
```
Comment (ron-murhammer) [inline src/games/strategy/ui/Util.java]: Constants should be in all caps with underscores.

File: src/util/image/DecorationPlacer.java [hunk 2]
```diff
@@ -700,23 +700,8 @@ private void fillCurrentImagePointsBasedOnImageFolder(final boolean pointsAreExa
...
-            seaName = name;
-          } else {
-            return name;
-          }
-        } // if
-      } // while
-    } // while
-    return seaName;
+  private Optional<String> findTerritoryName(final Point p) {
+    return Util.findTerritoryName(p, m_polygons);
```
Comment (ron-murhammer) [inline src/util/image/DecorationPlacer.java]: Would it be better to just remove this private method all together and just call the util method inline?

File: src/util/image/PlacementPicker.java [hunk 3]
```diff
@@ -526,22 +525,7 @@ private void loadPlacements() {
...
-          if (name.endsWith("Sea Zone") || name.startsWith("Sea Zone")) {
-            seaName = name;
-          } else {
-            return name;
-          }
-        } // if
-      } // while
-    } // while
-    return seaName;
+    return Util.findTerritoryName(p, m_polygons, "there be dragons");
```
Comment (ron-murhammer) [inline src/util/image/PlacementPicker.java]: What is "there be dragons" string for?

**Comment by frigoref** [comment]
Review status: 0 of 16 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/util/image/PlacementPicker.java, line 528 \[r1\]](https://reviewable.io:443/reviews/triplea-game/triplea/659#-KHF99LFXUaicFmrVdsV-r1-528:-KHHV1PxEHxjyfTmgeBr:-1725998863) ([raw file](https://github.com/triplea-game/triplea/blob/91a8ac9d286c8314c08ef415f8ab129a810ed142/src/util/image/PlacementPicker.java#L528)):*
<details><summary><i>Previously, ron-murhammer (Ron) wrote…</i></summary>
> What is "there be dragons" string for?</details>
I also didn't understand so I left it for someone with more knowledge to figure out ...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/triplea-game/triplea/659)*

**Pull request merged**